src/app/features/vendor/vendor-creation/creation.component.ts

```diff
@@ -1,10 +1,178 @@
 import { Component } from '@angular/core';
-// import { CommonModule } from '@angular/common';
+import { Vendor } from '../models/vendor.model';
+import { ConfirmationService, MessageService } from 'primeng/api';
+
+// LLD
+
+// This component is to add new vendors
+
+// it has the fields:
+
+// Profile Info:
+// Vendor Name: Required text input.
+// State: Required input Dropdown with state options.
+// Country: Required input Dropdown with country options.
+// Markets: Required input Multi-select dropdown with  options.
+// Contact Info:
+// Email: Required email input.
+// Phone: Required Input with country code dropdown.
+// Website: Optional text input.
+// Service Category:
+// Service: Required input Dropdown with  options.
+
+// It has the buttons:
+// Submit for approval
+// Cancel
+
+// This screen contains following actions:
+// Input values to different fields.
+// Submitting the data and going to View Profile page.
+// Cancelling the changes made and going back to Vendor Listing Page.
+
+// Submit:
+// When the user clicks on the submit for approval button a modal wil appear asking the confirmation  for data submission,
+// if user confirms, user will be directed to view profile page.
+
+// Cancel:
+// When the user clicks on cancel button a modal will appear asking confirmation for cancelling the data entered without submitting,
+// if user confirms , user will be directed back to Vendor Listing page.
 
 @Component({
   selector: 'app-creation',
   standalone: false,
   templateUrl: './creation.component.html',
   styleUrl: './creation.component.scss',
+  providers: [ConfirmationService,MessageService]
 })
-export class CreationComponent {}
+
+export class CreationComponent {
+  
+  // addVendorForm!: FormGroup;
+  vendor: Vendor[] | undefined;
+VendorName:string | undefined;
+emailAddress:string | undefined;
+selectedCode:string | undefined;
+phoneNumber:string | undefined;
+websiteName:string | undefined;
+selectedCountry:string | undefined;
+selectedMarket:string[] | undefined;
+selectedService:string | undefined;
```

Comment:
Why undefined is mentioned as a type?
Do we using those variables here?

Reply:
its removed